### PR TITLE
[UPD] Add control iterators speedup to sdp_optimize + adapt benchmark…

### PR DESCRIPTION
**Pull request by trigaut**
This PR divides by 2 or 3 the execution time of sdp_optimize (in both cases)

### CODE REVIEW

File: examples/stock-example.jl [hunk 1]
```diff
@@ -42,7 +42,7 @@ const S0 = 0.5
 proba = 1/N_XI*ones(N_XI) # uniform probabilities
 xi_support = collect(linspace(XI_MIN,XI_MAX,N_XI))
 xi_law = NoiseLaw(xi_support, proba)
-xi_laws = NoiseLaw[xi_law for t in 1:N_STAGES-1] 
+xi_laws = NoiseLaw[xi_law for t in 1:N_STAGES-1]
```
Comment (frapac) [inline examples/stock-example.jl:45]: Well done!

File: src/SDPoptimize.jl [hunk 2]
```diff
@@ -237,6 +237,8 @@ function sdp_solve_DH(model::StochDynProgModel,
     #Compute cartesian product spaces
     product_states, product_controls = generate_grid(model, param)
 
+    product_controls = collect(product_controls)
```
Comment (frapac) [inline src/SDPoptimize.jl:240]: I do not understand why is it faster, but ok^^

Comment (trigaut) [inline src/SDPoptimize.jl:240]: Me neither ^^.
I have to test if it does the same thing with product_states on instances with large time steps set (on small examples there is no difference).

**Comment by frapac** [comment]
I agree to merge

**Pull request merged**